config/config.go

```diff
@@ -14,10 +14,14 @@ type GoogleCloudStorageConfig struct {
 	Bucket                string `yaml:"bucket"`
 	UseDefaultCredentials bool   `yaml:"use_default_credentials"`
 	JSONCredentialsFile   string `yaml:"json_credentials_file"`
+	DisableReads          bool   `yaml:"disable_reads"`
```

Comment:
just curious, what is the use case for disable_reads?

Reply:
Write through from a cache that is less likely to get poisoned to one that is more likely to: for example having the cache shared by CI write through to the cache shared by developers.

The advantage over setting them up the other way around, with the developer shared cache reading through to the CI cache is that the developer cache doesn't need to wait for the transfers: for example if your CI cache is in a data center and your developer cache is in the office, the link between them might not be as fast as the link between developers and the developer cache. In that case waiting to download the artefact from the data center cache might be generally slower than building locally. But if you set up write through the data center cache can slowly send artefacts to the developer cache without blocking any builds.

Reply:
```
Developer machine <===> developer cache <---> CI cache
                    ^                     ^
            fast connection        slow connection
```

Developer machines can quickly read and write to developer cache, developer cache can slowly write to CI cache without blocking developers because uploads are asynchronous, but it cannot slowly read from CI cache without blocking developers because downloads are synchronous.